diff --git a/node_modules/expo-av/android/CMakeLists.txt b/node_modules/expo-av/android/CMakeLists.txt
index e55dbb2..f943917 100644
--- a/node_modules/expo-av/android/CMakeLists.txt
+++ b/node_modules/expo-av/android/CMakeLists.txt
@@ -43,6 +43,7 @@ target_link_libraries(
         ${LOG_LIB}
         fbjni::fbjni
         ReactAndroid::jsi
-        ReactAndroid::reactnativejni
         android
 )
+
+
diff --git a/node_modules/expo-av/android/src/main/java/expo/modules/av/ViewUtils.kt b/node_modules/expo-av/android/src/main/java/expo/modules/av/ViewUtils.kt
index d18b58a..dda344d 100644
--- a/node_modules/expo-av/android/src/main/java/expo/modules/av/ViewUtils.kt
+++ b/node_modules/expo-av/android/src/main/java/expo/modules/av/ViewUtils.kt
@@ -17,17 +17,7 @@ object ViewUtils {
 
   @UiThread
   private fun tryRunWithVideoViewOnUiThread(moduleRegistry: ModuleRegistry, viewTag: Int, callback: VideoViewCallback, promise: Promise) {
-    try {
-      val videoWrapperView = moduleRegistry.getModule(UIManager::class.java).resolveView(viewTag) as VideoViewWrapper?
-      val videoView = videoWrapperView?.videoViewInstance
-      if (videoView != null) {
-        callback.runWithVideoView(videoView)
-      } else {
-        promise.reject("E_VIDEO_TAGINCORRECT", "Invalid view returned from registry.")
-      }
-    } catch (e: IllegalViewOperationException) {
-      promise.reject("E_VIDEO_TAGINCORRECT", "Invalid view returned from registry.")
-    }
+    promise.reject("E_VIDEO_TAGINCORRECT", "Invalid view returned from registry.")
   }
 
   /**
@@ -48,17 +38,7 @@ object ViewUtils {
 
   @UiThread
   private fun tryRunWithVideoViewOnUiThread(moduleRegistry: ModuleRegistry, viewTag: Int, callback: VideoViewCallback, promise: expo.modules.kotlin.Promise) {
-    try {
-      val videoWrapperView = moduleRegistry.getModule(UIManager::class.java).resolveView(viewTag) as VideoViewWrapper?
-      val videoView = videoWrapperView?.videoViewInstance
-      if (videoView != null) {
-        callback.runWithVideoView(videoView)
-      } else {
-        promise.reject("E_VIDEO_TAGINCORRECT", "Invalid view returned from registry.", null)
-      }
-    } catch (e: IllegalViewOperationException) {
-      promise.reject("E_VIDEO_TAGINCORRECT", "Invalid view returned from registry.", null)
-    }
+    promise.reject("E_VIDEO_TAGINCORRECT", "Invalid view returned from registry.", null)
   }
 
   /**
diff --git a/node_modules/expo-av/android/src/main/java/expo/modules/av/video/FullscreenVideoPlayer.java b/node_modules/expo-av/android/src/main/java/expo/modules/av/video/FullscreenVideoPlayer.java
index ec4c842..24f64f8 100644
--- a/node_modules/expo-av/android/src/main/java/expo/modules/av/video/FullscreenVideoPlayer.java
+++ b/node_modules/expo-av/android/src/main/java/expo/modules/av/video/FullscreenVideoPlayer.java
@@ -11,7 +11,7 @@ import android.widget.FrameLayout;
 import java.lang.ref.WeakReference;
 
 import expo.modules.core.ModuleRegistry;
-import expo.modules.core.interfaces.services.KeepAwakeManager;
+
 import expo.modules.av.player.PlayerData;
 import expo.modules.kotlin.AppContext;
 
@@ -36,12 +36,10 @@ public class FullscreenVideoPlayer extends Dialog {
           AppContext appContext = fullscreenVideoPlayer.mAppContext.get();
           ModuleRegistry moduleRegistry = appContext != null ? appContext.getLegacyModuleRegistry() : null;
           if (moduleRegistry != null) {
-            KeepAwakeManager keepAwakeManager = moduleRegistry.getModule(KeepAwakeManager.class);
-            boolean keepAwakeIsActivated = keepAwakeManager != null && keepAwakeManager.isActivated();
-            if (isPlaying || keepAwakeIsActivated) {
+            if (isPlaying) {
               window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
             } else {
-              window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+              window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
             }
           }
         }
diff --git a/node_modules/expo-av/build/ExponentAV.web.js b/node_modules/expo-av/build/ExponentAV.web.js
index 4d8e8f0..14f5a17 100644
--- a/node_modules/expo-av/build/ExponentAV.web.js
+++ b/node_modules/expo-av/build/ExponentAV.web.js
@@ -1,4 +1,5 @@
-import { DeviceEventEmitter, PermissionStatus } from 'expo-modules-core';
+import { PermissionStatus } from 'expo-modules-core';
+import { DeviceEventEmitter } from 'react-native';
 import { RecordingOptionsPresets } from './Audio/RecordingConstants';
 async function getPermissionWithQueryAsync(name) {
     if (!navigator || !navigator.permissions || !navigator.permissions.query)
